foyer/xml_writer.py

```diff
@@ -95,6 +99,18 @@ def _write_atoms(self, root, atoms, forcefield, unique):
                 if key == 'doi':
                     label = eval(val)#[a for a in label]
                     label = ','.join([a for a in label])
+                elif key == 'overrides':
+                # Only write overrides atomtypes if they are in self.atom_type
+                    label = []
+                    for item in eval(val):
```

Comment:
This is largely unrelated to this PR but what _necessitated_ the use of `eval` here?

What is the shape of data that would raise `AttributeErrors` and `KeyErrors`?

Reply:
To be honest, I am not really sure. Here I am just kind of follow the earlier convention in this block of code. 
I will try strip off the eval part and see if it makes any difference.

Reply:
So here is what happened when I use `for item in val:`

```<Type name="opls_145" class="CA" element="C" mass="12.01078" def="[C;X3;r6]1[C;X3;r6][C;X3;r6][C;X3;r6][C;X3;r6][C;X3;r6]1" overrides="f,o,r,c,e,f,i,e,l,d,.,a,t,o,m,T,y,p,e,O,v,e,r,r,i,d,e,s,[,n,a,m,e,]" desc="" doi="10.1021/ja9621760"/>```

From the block above where we define `atomtype_attrs = collections.OrderedDict([key,val])`, `val` here is actually `forcefield.atomTypeOverrides[name]`, and hence we need the `eval` step to return a primitive data type.

Reply:
Ah I see. No need to hold up your current PR but a pattern for avoiding this would be to define `getter` functions for every attribute.

They would look roughly like `get_XYZ(atom)`. The override function for example could look like:

```python
def get_atomtype_overrides(atom):
    return forcefield.atomTypeOverrides[atom.name]
```

The loop would be rewritten as:
```python
for attribute, getter in atomtype_attribute_getters.items():
   if attribute == 'overrides':
        for item in getter(atom):
           ...
```